sdk/core/core-client/test/serializer.spec.ts

```diff
@@ -538,7 +538,7 @@ describe("Serializer", function () {
 
     it("should correctly serialize object version of polymorphic discriminator", function () {
       const serializer = createSerializer(Mappers);
```

Comment:
can we also add some additional tests? This code is really complex and has had [tricky to debug issues before](https://github.com/Azure/azure-sdk-for-js/pull/19455)